src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -76,21 +73,69 @@ public IndexStore(DocumentDatabase documentDatabase, ServerStore serverStore)
         {
             _documentDatabase = documentDatabase;
             _serverStore = serverStore;
-            _logger = LoggingSource.Instance.GetLogger<IndexStore>(_documentDatabase.Name);
+            Logger = LoggingSource.Instance.GetLogger<IndexStore>(_documentDatabase.Name);
 
             var stoppedConcurrentIndexBatches = _documentDatabase.Configuration.Indexing.NumberOfConcurrentStoppedBatchesIfRunningLowOnMemory;
             StoppedConcurrentIndexBatches = new SemaphoreSlim(stoppedConcurrentIndexBatches);
         }
 
-        public void HandleDatabaseRecordChange(DatabaseRecord record, long index)
+        public void HandleDatabaseRecordChange(DatabaseRecord record, long raftIndex)
         {
             if (record == null)
                 return;
 
-            HandleSorters(record, index);
-            HandleDeletes(record, index);
-            HandleChangesForStaticIndexes(record, index);
-            HandleChangesForAutoIndexes(record, index);
+            var indexesToStart = new List<Index>();
+
+            HandleSorters(record, raftIndex);
+            HandleDeletes(record, raftIndex);
+
+            HandleChangesForStaticIndexes(record, raftIndex, indexesToStart);
+            HandleChangesForAutoIndexes(record, raftIndex, indexesToStart);
+
+            if (indexesToStart.Count <= 0)
+                return;
+
+            var sp = Stopwatch.StartNew();
+
+            if (Logger.IsInfoEnabled)
+                Logger.Info($"Starting {indexesToStart.Count} indexes");
+
+            Parallel.ForEach(indexesToStart, index =>
+            {
```

Comment:
It is CPU bounded, no? If it is we might what to set `MaxDegreeOfParallelism` to the number of cores.

Reply:
Not sure if we should not take 50% of the cores or 75%. Thoughts?

Reply:
+1 for 50%

Reply:
Remember to handle a single core scenario as well.
But let's not use the cores on the machine, use the _licensed_ cores.

Reply:
But all of licensed ones or 50%?

Reply:
Math.Max(1, LicensedCores / 2)

Reply:
Actually Math.Max(1, UtilizedCores / 2)